persistent/cPersistence.c

```diff
@@ -180,13 +181,57 @@ ghostify(cPersistentObject *self)
         _estimated_size_in_bytes(self->estimated_size);
     ring_del(&self->ring);
     self->state = cPersistent_GHOST_STATE;
+
+    /* clear __dict__ */
     dictptr = _PyObject_GetDictPtr((PyObject *)self);
     if (dictptr && *dictptr)
     {
         Py_DECREF(*dictptr);
         *dictptr = NULL;
     }
 
+    /* clear all slots besides _p_* */
+    slotnames = pickle_slotnames(Py_TYPE(self));
+    if (slotnames && slotnames != Py_None)
+    {
+        int i;
+
+        for (i = 0; i < PyList_GET_SIZE(slotnames); i++)
+        {
+            PyObject *name;
+            char *cname;
+            int is_special;
+
+            name = PyList_GET_ITEM(slotnames, i);
+#ifdef PY3K
+            if (PyUnicode_Check(name))
+            {
+                PyObject *converted = convert_name(name);
+                cname = PyBytes_AS_STRING(converted);
+#else
+            if (PyBytes_Check(name))
+            {
+                cname = PyBytes_AS_STRING(name);
+#endif
+                is_special = !strncmp(cname, "_p_", 3);
+#ifdef PY3K
+                Py_DECREF(converted);
+#endif
+                if (is_special) /* skip persistent */
+                {
+                    continue;
+                }
+            }
+
+            /* NOTE: this skips our delattr hook */
+            if (PyObject_GenericSetAttr((PyObject *)self, name, NULL) < 0)
```

Comment:
Unless my eyes are fooling me, because it is outside the look, this only clears the *last* slot's value.